src/lib.rs

```diff
@@ -5,94 +5,109 @@ use std::ops::Add;
 use std::fmt;
 
 /// Represents a pointer to a field of type `U` within the type `T`
+#[repr(transparent)]
 pub struct FieldOffset<T, U>(
     /// Offset in bytes of the field within the struct
     usize,
     /// A pointer-to-member can be thought of as a function from
     /// `&T` to `&U` with matching lifetimes
-    PhantomData<for<'a> Fn(&'a T) -> &'a U>
+    PhantomData<dyn for<'a> Fn(&'a T) -> &'a U>
```

Comment:
Why this change?

Reply:
Omitting `dyn` here is a warning now. It doesn't change the meaning of the code.